src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala

```diff
@@ -617,22 +619,113 @@ object Helpers extends SparkSessionWrapper {
    *                        or direct access via s3:// or abfss:// or dbfs:/mnt/ etc.
    * @return
    */
+//  def registerRemoteOverwatchIntoLocalMetastore(
+//                                                 remoteWorkspace: Workspace,
+//                                                 localDataTarget: DataTarget
+//                                               ): Seq[WorkspaceMetastoreRegistrationReport] = {
+//
+//    val newConfigParams = remoteWorkspace.getConfig.inputConfig.copy(dataTarget = Some(localDataTarget))
+//    val newConfigArgs = JsonUtils.objToJson(newConfigParams).compactString
+//    val localTempWorkspace = Initializer(newConfigArgs, disableValidations = true)
+//    val registrationReport = localTempWorkspace.addToMetastore()
+//    val b = Bronze(localTempWorkspace, suppressReport = true, suppressStaticDatasets = true)
+//    val g = Gold(localTempWorkspace, suppressReport = true, suppressStaticDatasets = true)
+//
+//    b.refreshViews()
+//    g.refreshViews()
+//    registrationReport
+//
+//  }
   def registerRemoteOverwatchIntoLocalMetastore(
-                                                 remoteWorkspace: Workspace,
-                                                 localDataTarget: DataTarget
-                                               ): Seq[WorkspaceMetastoreRegistrationReport] = {
+                                                 remoteStoragePrefix: String,
+                                                 remoteWorkspaceID: String,
+                                                 localETLDatabaseName: String = "",
+                                                 localConsumerDatabaseName: String = "",
+                                                 remoteETLDataPathPrefixOverride: String = "",
+                                                 usingExternalMetastore: Boolean = false,
+                                                 workspacesAllowed: Array[String] = Array()
+                                               ): Seq[WorkspaceMetastoreRegistrationReport]  = {
+
+    // Derive eltDatapathPrefix
+    val eltDataPathPrefix = if (remoteETLDataPathPrefixOverride == "") {
+      remoteStoragePrefix + "/global_share"
+    } else remoteETLDataPathPrefixOverride
+
+    // Check whether eltDatapathPrefix Contains PipReport
+    val pipReportPath = eltDataPathPrefix+"/pipeline_report"
+    try{
+      dbutils.fs.ls(pipReportPath)
+      logger.log(Level.INFO, s"${pipReportPath} is a Delta table.....Proceed")
```

Comment:
to confirm that it's a delta table you shouldn't just verify the presence of the parent folder but also that `_delta_log` folder.

dbutils.fs.ls(pipReportPath) is not a valid test. I think you could test that it's `nonEmpty` 
`dbutils.fs.ls(s"$pipReportPath/_delta_log").nonEmpty`

Reply:
oh I see -- i remember now that fs.ls does throw an exception on empty dir either way is fine, let's just verify the delta_log path not the parent path. :)